### PR TITLE
[VPAT] Adds the ability to click elements with onclick attributes using they keyboard

### PR DESCRIPTION
**Pull request by elihschiff**
### Please check if the PR fulfills these requirements:

* [X] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?

There are a lot of html elements on Submitty that use onclick="myJSfunction()" but you are sometimes unable to "click" them using a keyboard.
#4909 (this issue should not be closed as there is still the problem of draggable elements that I do not have any idea how to fix)

### What is the new behavior?

This pr adds a js class `key_to_click` which when added to any element with an onclick will allow a user to use the ENTER or SPACE keys to click the element. I also added it to every onclick element I could find.

This pr also adds tabindex="0" to every element that has onclick

### CODE REVIEW

**Comment by elihschiff** [comment]
This also solves "Onclick should also have keyboard option" in #4491

File: site/app/templates/autograding/AutoChecks.twig [hunk 1]
```diff
@@ -48,7 +48,7 @@
         <h4>
             {{ element.title }}
             {% if element.show_popup %}
-                <span onclick="openPopUp('{{ popup_css_file }}', '{{ element.title }}', {{ index }}, {{ check_index }}, {{ side }});">
+                <span onclick="openPopUp('{{ popup_css_file }}', '{{ element.title }}', {{ index }}, {{ check_index }}, {{ side }});" class="key_to_click" tabindex="0">
```
Comment (hdrodz97) [inline site/app/templates/autograding/AutoChecks.twig]: Button doesn't respond to enter key on Firefox

File: site/app/templates/autograding/AutoChecks.twig [hunk 2]
```diff
@@ -10,7 +10,7 @@
         <br>
         {% if check.actual %}
         <div style="height:23px; float:right;">
-            <a id="show_char_{{ index }}_{{ loop.index0 }}" class="btn btn-default" style="float:right;" onclick="changeDiffView('container_{{ index }}_{{ loop.index0 }}', '{{ gradeable_id }}', '{{ who }}', '{{ display_version }}', '{{ index }}', '{{ loop.index0 }}', 'white_space_helper_{{ index }}')">Visualize whitespace characters</a>
+            <a id="show_char_{{ index }}_{{ loop.index0 }}" class="btn btn-default key_to_click" tabindex="0" style="float:right;" onclick="changeDiffView('container_{{ index }}_{{ loop.index0 }}', '{{ gradeable_id }}', '{{ who }}', '{{ display_version }}', '{{ index }}', '{{ loop.index0 }}', 'white_space_helper_{{ index }}')">Visualize whitespace characters</a>
```
Comment (hdrodz97) [inline site/app/templates/autograding/AutoChecks.twig]: Button doesn't respond to enter key on Firefox

File: site/app/templates/grading/electronic/Status.twig [hunk 3]
```diff
@@ -167,10 +167,10 @@
     {% else %}
         <div class = "tab">
             <div class =tab-bar-wrapper">
-                <a class="nav-bar normal-btn"  onclick="openStat(event, 'numerical-data'), this.blur()" id="defaultOpen">Data</a>
-                <a class="nav-bar normal-btn"  id="btn-hist-auto" onclick="openStat(event, 'autograding-score-histogram'), this.blur()">Autograding Histogram</a>
-                <a class="nav-bar normal-btn"  id="btn-comp" onclick="openStat(event, 'component-averages'), this.blur()">Manual Grading Component Averages</a>
-                <a class="nav-bar normal-btn"  id="btn-hist-tot" onclick="openStat(event, 'total-score-histogram'), this.blur()">Overall Histogram</a>
+                <a class="nav-bar normal-btn key_to_click" tabindex="0"  onclick="openStat(event, 'numerical-data'), this.blur()" id="defaultOpen">Data</a>
```
Comment (hdrodz97) [inline site/app/templates/grading/electronic/Status.twig]: On Firefox, this properly opens the tab but it doesn't re-highlight it as it does when clicked

File: site/public/templates/grading/Component.twig [hunk 4]
```diff
@@ -34,7 +34,7 @@ Required Input:
 
     {# Title bar (only part that triggers collapse on click) #}
     <div class="box-title container">
-        <div class="row header-block" onclick="{% block component_click %}{% endblock %}">
+        <div class="row header-block key_to_click" tabindex="0" onclick="{% block component_click %}{% endblock %}">
```
Comment (hdrodz97) [inline site/public/templates/grading/Component.twig]: Doesn't respond to enter key on Firefox

File: site/public/templates/grading/EditComponent.twig [hunk 5]
```diff
@@ -97,14 +97,14 @@
...
                            name="count-type-{{ component.id }}" {{ component.default == 0 ? 'checked' : '' }}
                            onclick="onClickCountUp(this)" id="grade_by_count_up"> <label for="grade_by_count_up">Grade by Count Up (from zero)</label>
                 </span>
             </div>
             <div class="col">
                 <span class="radio-input">
-                    <input type="radio" class="count-down-selector count-type-selector"
+                    <input type="radio" class="count-down-selector count-type-selector key_to_click" tabindex="0"
                            name="count-type-{{ component.id }}" {{ component.default != 0 ? 'checked' : '' }}
                            onclick="onClickCountDown(this)" id="grade_by_count_down"> <label for="grade_by_count_down">Grade by Count Down (from "Points")</label>
```
Comment (hdrodz97) [inline site/public/templates/grading/EditComponent.twig]: Not *entirely* sure whether this is necessary; it seems like when radio buttons are concerned, one group counts as one "entry" in the tab ordering and switching between items is accomplished with the arrow keys.

File: site/public/templates/grading/Mark.twig [hunk 6]
```diff
@@ -9,7 +9,7 @@ Required Input:
     show_publish: True to show the 'publish' option
 #}
 <div id="mark-{{ mark.id }}"
-     class="mark-container row {{ mark.publish ? 'mark-publish' : '' }} {{ first_mark ? 'mark-first' : '' }} {{ not edit_marks_enabled and mark_disabled ? 'mark-disabled' : '' }} {{ mark.deleted ? 'mark-deleted' : '' }}"
+     class="mark-container key_to_click row {{ mark.publish ? 'mark-publish' : '' }} {{ first_mark ? 'mark-first' : '' }} {{ not edit_marks_enabled and mark_disabled ? 'mark-disabled' : '' }} {{ mark.deleted ? 'mark-deleted' : '' }}" tabindex="0"
```
Comment (hdrodz97) [inline site/public/templates/grading/Mark.twig]: These sorts of headers in general don't seem to work on Firefox with the Enter key.

File: site/public/templates/grading/Mark.twig [hunk 7]
```diff
@@ -78,7 +78,7 @@ Required Input:
             </span>
         {% endif %}
     {% endif %}
-    <span class="col-no-gutters mark-tools" onclick="onGetMarkStats(this);event.stopPropagation()">
+    <span class="col-no-gutters mark-tools key_to_click" tabindex="0" onclick="onGetMarkStats(this);event.stopPropagation()">
```
Comment (hdrodz97) [inline site/public/templates/grading/Mark.twig]: Button doesn't respond to enter on Firefox

**Comment by bmcutler** [comment]
@elihschiff 
Please resolve conflicts :)

**Pull request merged**